### ISSUE TITLE
DynamoDB Auto Scaling

### ISSUE THREAD

**Issue by derekmceachern** [awslabs/emr-dynamodb-connector#28]
Now that DynamoDB supports Auto Scaling is there any chance this can be updated to periodically go back and check the read/write capacity units and utilize any additional capacity added?

**Comment by niranjv** [comment]
+1

**Comment by canada4663** [comment]
+1

**Comment by mattklein** [comment]
Beware though, that could result in a feedback loop whereby DynamoDB would increase capacity, then the connector would increase its utilization, then DynamoDB would increase capacity again because of the increased utilization, until capacity reached the max possible. This might not be desired behavior.

**Comment by derekmceachern** [comment]
@mattklein Yes, this could be an issue if you don't want this to happen. In my case though it is exactly what I want.

**Comment by niranjv** [comment]
@mattklein This is exactly what I want too. Max capacity can be set in DynamoDB console, so it's not going to increase without limit. The number of times per day capacity can increase is unlimited.

When incoming reads/writes decrease, capacity also decreases accordingly. But the number of times per day capacity can decrease is limited. This can result in having unnecessary idle capacity until the next day when capacity can be decreased again. 

You can see this behavior in my auto-scaling test at https://github.com/niranjv/aws/blob/master/dynamodb/autoscaling-test/dynamodb-autoscaling-write-capacity.png where I imported ~ 35 million items into a table using [batch_writer in boto3](http://boto3.readthedocs.io/en/latest/reference/services/dynamodb.html#DynamoDB.Table.batch_writer) starting with write capacity of 5 units and max capacity of 10,000 units.

**Comment by weikai2** [comment]
+1
It's painful to have to tweak minimum provisioned capacity setting each time when I run an export. Maybe simply allowing user to pass in a fixed number of capacity is what most people want. (with a clearly stated risk that the increased capacity could take more time than expected to go down when the job finishes)

**Comment by taklwu** [comment]
https://github.com/awslabs/emr-dynamodb-connector/commit/5f67329554755ec81c93d807b9c0765b85150b8d should have supported, please compiled the repo and replace those jars on the EMR cluster.

**Comment by BStokes** [comment]
This is still an issue when running GlueJobs especially for ingress to DDB.  There needs to be a check for the write capacity from the 'application-autoscaling' service (MaxCapacity for the dynamodb:table:WriteCapacityUnits dimension), and not check the provisioned to determine if it set or not (per [5f67329](https://github.com/awslabs/emr-dynamodb-connector/commit/5f67329554755ec81c93d807b9c0765b85150b8d)).  Reason being [read and write capacity] provision is set to the min/starting point of autoscale and not null.  Additionally, choosing the min in this situation is incorrect and needs to be max allowable to scale.
